src/main/java/lotto/domain/WinningBall.java

```diff
@@ -0,0 +1,25 @@
+package lotto.domain;
+
+public class WinningBall {
+
+  private final LottoBalls winLottoBalls;
+  private final LottoBall bonusBall;
+
+  public WinningBall(LottoBalls winLottoBalls, LottoBall bonusBall) {
+    this.winLottoBalls = winLottoBalls;
+    this.bonusBall = bonusBall;
+  }
+
+  public Ranking calculateRank(LottoBalls drawBalls) {
+    int count = 0;
+    count = drawBalls.countContainingWinNumbers(winLottoBalls);
+    if (isNoMatchBonusSecond(drawBalls, count)) {
+      return Ranking.SECOND;
+    }
+    return Ranking.ofCount(count);
+  }
+
+  private boolean isNoMatchBonusSecond(LottoBalls drawBalls, int count) {
+    return count == 5 && !drawBalls.containBonusBall(bonusBall);
+  }
```

Comment:
해당 로직은 Rank에서 수행하면 어떨까요 :) ?
```suggestion
    return Raking.find(count, drawBalls.containBonusBall(bonusBall));
```

Reply:
네.  당첨공을 몇개 가지고 있는지만 winningBall 에서 계산하고 , 등수 계산은 Ranking 에게 맡겨도 되겠네요!